### PR TITLE
feat: convert server plugin and support server plugin params

### PR DESCRIPTION
**Pull request by zllkjc**
# PR Details

convert @modern-js/plugin-server and @modern-js/plugin-polyfill packages server part.

now server can load plugin by itself.

## Types of changes

- [ ] Docs change / refactoring / dependency upgrade
- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist

- [x] My code follows the code style of this project.
- [x] I have updated changeset
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have read the **CONTRIBUTING** document.
- [x] I have added tests to cover my changes.
- [ ] All new and existing tests passed.

### CODE REVIEW

File: packages/server/core/src/loadPlugins.ts [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+export const loadPlugins = (plugins: Plugin[], appDirectory: string) => {
+  const resolvePlugin = (p: Plugin) => {
+    const isPluginInstance = typeof p !== 'string' && !Array.isArray(p);
+    if (isPluginInstance) {
+      return {
+        module: p,
+      };
+    }
+
+    const [pkg, options] = typeof p === 'string' ? [p, undefined] : p;
```
Comment (chenjiahan) [inline packages/server/core/src/loadPlugins.ts:28]: server 插件在哪里定义 options 嘞

Comment (zllkjc) [reply packages/server/core/src/loadPlugins.ts:28]: app-tools 会把 core 解析出来的 server 插件配置都透传过来，就和 cli 定义方式一样

Comment (zllkjc) [reply packages/server/core/src/loadPlugins.ts:28]: 上面说的 core 是 cli 的 core

File: packages/cli/core/src/loadPlugins.ts [hunk 2]
```diff
@@ -126,9 +126,7 @@ export const loadPlugins = (
 
     // server plugins don't support to accept params
     if (server && typeof server === 'string') {
-      const path = tryResolve(server, appDirectory);
-
-      loadedPlugin.server = { pluginPath: path };
+      loadedPlugin.server = server;
```
Comment (yimingjfe) [inline packages/cli/core/src/loadPlugins.ts:129]: 这个字段是不是其实不需要了？ serverPkg 跟这个字段是一样的

**Review by chenjiahan — APPROVED** [review]

**Pull request merged**